src/machines/java/com/enderio/machines/common/item/FluidTankItem.java

```diff
@@ -1,17 +1,41 @@
 package com.enderio.machines.common.item;
 
+import com.enderio.EnderIO;
+import com.enderio.core.client.item.ItemTooltip;
 import com.enderio.machines.client.rendering.item.FluidTankBEWLR;
+import com.enderio.machines.common.block.MachineBlock;
 import net.minecraft.client.renderer.BlockEntityWithoutLevelRenderer;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
 import net.minecraft.world.item.BlockItem;
-import net.minecraft.world.level.block.Block;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
 import net.minecraftforge.client.extensions.common.IClientItemExtensions;
+import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
 import net.minecraftforge.common.util.NonNullLazy;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import net.minecraftforge.fluids.capability.templates.FluidHandlerItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
+import java.util.List;
+import java.util.Optional;
 import java.util.function.Consumer;
 
 public class FluidTankItem extends BlockItem {
-    public FluidTankItem(Block block, Properties properties) {
+
+    protected final int capacity;
+    public FluidTankItem(MachineBlock block, Properties properties, int capacity) {
         super(block, properties);
+        this.capacity = capacity;
+    }
+
+    @Override
+    public int getMaxStackSize(ItemStack stack) {
+        return 1;//TODO: when fluid tank entity accepts item stacks of more than 1 in the internalDrain/fill. Remove this method to allow fluid tank items to stack to 64.
```

Comment:
Why would we allow fluid tank stacking?

Reply:
It's a feature in 1.12 and is one of my absolute favourite features of EnderIO. It makes the tanks of enderIO the most powerful fluid storing items in modded minecraft in my opinion!